src/proxy_channel.cc

```diff
@@ -85,6 +93,7 @@ ProxyHandlerResult ProxyService::handleTrigger(ProxyTrigger triggerRaw) {
   if (trigger->fields.type & TriggerSync) {
```

Comment:
How about we just say `trigger->fields.type & TriggerSync || inflightRequests > MAX_INFLIGHT_REQUEST` here to merge the same block above

Reply:
Current way might have performance issue. When we call `flush`, it does busy polling until all inflight requests appear in CQ. This might waste some CPU cycles which instead could be used for submitting new requests. A better logic could be:
```
while (proxy_thread_running) {
  if (there_are_new_requests_to_submit) submit_requests();
  inflight += num_submitted_requests;
  if (there_are_new_requests_completed) complete_requests();
  inflight -= num_completed_requests;
}
```
But current `connection::flush` api doesn't do this. Perhaps we need to expose something like https://github.com/microsoft/mscclpp/blob/main/src/ib.cc#L288 to proxy thread.